designs/osc_high_availability/high_availability.md

```diff
@@ -0,0 +1,100 @@
+# OSC High Availability
+- OSC High Availability eliminates from single points of failure which protects from unplanned downtime.
+- Performs regular checks to make sure each OSC Service Node is working properly, and if one fails, switch it out for one that is working.
+- It tries to bring back the failed service/nodes into the cluster and maintains the same number of service nodes.
+    
+#### There are 2 types of failures
+1. Service Stops
+	OSC service may be stopped abnormaly.
+2. Node Level Stops
+	OSC Service Node Stopped because of some reasons.
+
+## Background
+Currently, OSC does not have the HA configuration option to handle above failure types. So, A high availability OSC will work by having more service nodes than it needs, will perform regular checks to make sure each service/node is working properly. If current active node moves into failure state then it turns on the service in other working node.
+
+We have explored three approaches.
+1. OSC node cluster with Active/Passive setup using Pacemaker, Corosync and Distributed Replicated Block Device(DRBD).
+2. Segregate Application and Persistence Layers(The option has discarded since H2 does not provide the option of Active/Passive Cluster)
+3. Using NFS and Custom Monitor Script
+
+## OSC HA with Active/Passive Setup 
+OSC node cluster with Active/Passive setup using Pacemaker, Corosync and Distributed Replicated Block Device(DRBD).
```

Comment:
break into logical paragraphs. Again looks like formatting was done in the raw format, but it does not translate to the final markdown output.